docs/src/starks/protocol_overview.md

```diff
@@ -1,87 +1,149 @@
-# Protocol overview
-TODO
+# Protocol Overview
 
-## Why use roots of unity?
+In this section we start diving deeper before showing the formal protocol. If you haven't done so, we recommend first reading the "Recap" section.
 
-Whenever we interpolate or evaluate trace, boundary and constraint polynomials, we use some $2^n$-th roots of unity. There are a few reasons for this:
+At a high level the protocol works as follows. The starting point is a matrix $T$ that encodes the trace of a valid execution of the program. This matrix needs to be in a special format so that its correctness is equivalent to checking a finite number of polynomial equations on its rows. Transforming the execution to this matrix is what's called the arithmetization process.
```

Comment:
```suggestion
At a high level the protocol works as follows. The starting point is a matrix $T$ that encodes the trace of a valid execution of the program. This matrix needs to be in a special format so that its correctness is equivalent to checking a finite number of polynomial equations on its rows. Transforming the execution to this matrix is what's called the arithmetization process.
```